indico/modules/events/registration/client/js/form/fields/ShowIfInput.jsx

```diff
@@ -0,0 +1,86 @@
+// This file is part of Indico.
+// Copyright (C) 2002 - 2025 CERN
+//
+// Indico is free software; you can redistribute it and/or
+// modify it under the terms of the MIT License; see the
+// LICENSE file for more details.
+
+import PropTypes from 'prop-types';
+import React, {useState} from 'react';
+import {useForm} from 'react-final-form';
+import {useSelector} from 'react-redux';
+
+import {FinalDropdown} from 'indico/react/forms';
+import {Fieldset} from 'indico/react/forms/fields';
+import {Translate} from 'indico/react/i18n';
+
+import {getItems} from '../selectors';
+
+import {getFieldRegistry} from './registry';
+
+export function ShowIfInput({hasValueSelected}) {
+  const items = Object.values(useSelector(getItems));
+  const fieldRegistry = getFieldRegistry();
+  const form = useForm();
+  const [showValue, setShowValue] = useState(hasValueSelected);
+  const [, setSelectedField] = useState(null);
+  let options = [];
+  const {showIfFieldId, id: thisFieldId} = form.getState().values;
+  if (showValue) {
+    const [field] = items.filter(({id}) => id === showIfFieldId);
```

Comment:
```suggestion
    const field = items.find(({id}) => id === showIfFieldId);
```

Reply:
Done in [c2a5d11](https://github.com/indico/indico/pull/6678/commits/c2a5d113071ba5093f7f0b9fadc53f371f6d3fb9).